pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -214,14 +214,22 @@ func (r *ReconcileKnativeServing) install(instance *servingv1alpha1.KnativeServi
 	instance.Status.Version = version.Version
 	log.Info("Install succeeded", "version", version.Version)
 	instance.Status.MarkInstallSucceeded()
-	r.exposeMetrics()
 	return nil
 }
 
 // Expose metrics for installed knative serving operator
-func (r *ReconcileKnativeServing) exposeMetrics() {
-	log.Info("expose metrics for installed knative serving")
-	knativeVersion.WithLabelValues(version.Version).Set(float64(time.Now().Unix()))
+func (r *ReconcileKnativeServing) exposeMetrics(instance *servingv1alpha1.KnativeServing) error {
+	if instance.Status.GetConditions() != nil {
+		log.Info("expose health status for installed knative serving")
+		if instance.Status.IsReady() {
```

Comment:
You can deduplicate a fair amount by doing:

```golang
status := 0
if instance.Status.IsReady() {
    status = 1
}
servingHealth.WithLabelValues(strconv.FormatBool(instance.Status.IsDependenciesInstalled()),
    strconv.FormatBool(instance.Status.IsAvailable()), strconv.FormatBool(instance.Status.IsInstalled())).Set(status)
```

Assuming you take the suggestion above.

Reply:
Yep Thank you
Updated changes